griptape/engines/rag/modules/base_rag_module.py

```diff
@@ -17,3 +17,12 @@ def generate_query_prompt_stack(self, system_prompt: str, query: str) -> PromptS
         return PromptStack(
             messages=[Message(system_prompt, role=Message.SYSTEM_ROLE), Message(query, role=Message.USER_ROLE)]
         )
+
+    def context_param(self, context: RagContext, key: str) -> Optional[Any]:
+        return context.module_params.get(self.name, {}).get(key)
+
+    def set_context_param(self, context: RagContext, key: str, value: Any) -> None:
+        if not isinstance(context.module_params.get(self.name), dict):
```

Comment:
Should this be `if self.name not in context.module_params`? Using `isinstance` feels inappropriate but maybe I'm missing something.

Reply:
I am testing for presence and for type being explicitly a `dict`.